server/src/collection.rs

```diff
@@ -205,6 +205,30 @@ impl Collection {
         None
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            item.set_locked(locked).await?;
+        }
+
+        let manager = self.manager.lock().await;
+
+        if self.is_locked().await != locked {
+            self.locked
+                .store(locked, std::sync::atomic::Ordering::Relaxed);
+            let signal_emitter = manager.signal_emitter(&self.path)?;
+            self.locked_changed(&signal_emitter).await?;
+        }
+
+        let service_path =
+            OwnedObjectPath::try_from(oo7::dbus::api::Service::PATH.as_deref().unwrap()).unwrap();
```

Comment:
It is already an ObjectPath. Maybe the signal_emitter method should be changed to take anything that can be turned into an ObjectPath instead of having to always pass a &OwnedObjectPath.

Reply:
Done, thanks!